Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -152,12 +153,17 @@ namespace ROS2
         return Internal::GetFirstROS2FrameAncestor(GetEntity());
     }
 
-    const AZ::Transform& ROS2FrameComponent::GetFrameTransform() const
+    AZ::Transform ROS2FrameComponent::GetFrameTransform() const
     {
         auto* transformInterface = Internal::GetEntityTransformInterface(GetEntity());
-        if (GetParentROS2FrameComponent() != nullptr)
+        if (const auto* parentFrame = GetParentROS2FrameComponent(); parentFrame != nullptr)
         {
-            return transformInterface->GetLocalTM();
+            auto* ancestorTransformInterface = Internal::GetEntityTransformInterface(parentFrame->GetEntity());
+            AZ_Assert(ancestorTransformInterface, "No transform interface for an entity with a ROS2Frame component, which requires it!");
+
+            auto ancestorWorldTm = ancestorTransformInterface->GetWorldTM();
+            auto thisWorldTm = transformInterface->GetWorldTM();
+            return ancestorWorldTm.GetInverse() * thisWorldTm;
```

Comment:
suggestion: For these types of transforms I came across a wonderful article a few years ago with a great recommendation on naming variables for matrix transforms - https://www.sebastiansylvan.com/post/matrix_naming_convention/.

The general idea is to have the space encoded in the variable name, and have them match up when multiplying, so we could rename things below like so....


```
worldFromAncestor
worldFromThis

worldFromAncestor = ancestorFromWorld.GetInverse();
return ancestorFromWorld * worldFromThis; // here we can see the 'world' parts of the variables are next to each other, confirming this is the right multiplication order
```

Totally optional, feel free to ignore, but I always like mentioning it as I found it incredibly useful and tend to use that style where I can now.

Reply:
This increases clarity, thank you for such a good suggestion, I will apply it!

Reply:
I agree we could use tests for this feature. I would like to think about adding more Unit tests and map out the work for the Gem in general. I have created an issue to track unit tests for the Frame component specifically: https://github.com/o3de/o3de-extras/issues/178. For new features, we will be adding unit tests when possible.